thriftrw/spec/struct.pyx

```diff
@@ -265,6 +266,15 @@ cdef class StructTypeSpec(TypeSpec):
                     )
                 else:
                     continue
+
+            # Since we validate at construction time, child structs are
+            # almost certain valid unless consumers are directly mutating
+            # thrift structs. As an optimization, avoid recursively revalidating
+            # these.
+            if isinstance(field.spec, (UnionTypeSpec, StructTypeSpec)):
```

Comment:
might be better and faster to just check `field.spec.ttype_code == ttype.STRUCT`

would also get rid of cyclic import